spec/compiler/parser/parser_spec.cr

```diff
@@ -919,6 +918,17 @@ module Crystal
     assert_syntax_error "macro foo; {% foo = 1 }; end"
     assert_syntax_error "macro def foo : String; 1; end"
 
+    assert_syntax_error "macro foo=;end", "macro can't be a setter"
+    assert_syntax_error "macro Foo;end", "macro can't have a receiver"
+    assert_syntax_error "macro foo.bar;end", "macro can't have a receiver"
+    assert_syntax_error "macro Foo.bar;end", "macro can't have a receiver"
+    assert_syntax_error "macro foo&&;end"
+    assert_syntax_error "macro foo"
+
+    ["`", "<<", "<", "<=", "==", "===", "!=", "=~", "!~", ">>", ">", ">=", "+", "-", "*", "/", "//", "~", "%", "&", "|", "^", "**", "[]?", "[]=", "<=>", "&+", "&-", "&*", "&**"].each do |op|
+      assert_syntax_error "macro #{op};end", "only '[]' can be used as an operator macro"
```

Comment:
This error message is very cryptic. I'd just say that it's an invalid name for a macro. There's no need for a reference to `[]` when you're trying to define a completely different macro.